### PR TITLE
Hotfix contnames

### PR DESCRIPTION
**Pull request by mfasDa**

### CODE REVIEW

**Comment by mfasDa** [comment]
@saiola, @lcunquei, @raymondEhlers 

Please review changes proposed in the pull request.

**Comment by mfasDa** [comment]
+test

File: PWG/EMCAL/EMCALbase/AliMCParticleContainer.cxx [hunk 1]
```diff
@@ -240,6 +240,8 @@ const char* AliMCParticleContainer::GetTitle() const
 {
   static TString trackString;
 
+  trackString = TString::Format("%s_pT%04d", GetArrayName().Data(), static_cast<int>(GetMinPt()*1000.0));
+  /*
   if (GetMinPt() == 0) {
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALbase/AliMCParticleContainer.cxx]: I would suggest just removing the old code - it's always in the history if needed

File: PWG/EMCAL/EMCALbase/AliParticleContainer.cxx [hunk 2]
```diff
@@ -433,6 +435,8 @@ const char* AliParticleContainer::GetTitle() const
 {
   static TString trackString;
 
+  trackString = TString::Format("%s_pT%04d", GetArrayName().Data(), static_cast<int>(GetMinPt()*1000.0));
+/*
   if (GetMinPt() == 0) {
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALbase/AliParticleContainer.cxx]: Same here

File: PWG/EMCAL/EMCALbase/AliTrackContainer.cxx [hunk 3]
```diff
@@ -600,6 +600,8 @@ const char* AliTrackContainer::GetTitle() const
 {
   static TString trackString;
 
+  trackString = TString::Format("%s_pT%04d", GetArrayName().Data(), static_cast<int>(GetMinPt()*1000.0));
+  /*
   if (GetMinPt() == 0) {
```
Comment (raymondEhlers) [inline PWG/EMCAL/EMCALbase/AliTrackContainer.cxx]: And here

**Comment by aiola** [comment]
looks good to me!

**Comment by mfasDa** [comment]
@raymondEhlers @aiola 

Thanks for your comments! My main worries are about the handling when momentum determination fails. This was not handled up to now at all, and I think if GetMomentum returns false a track / particle / cluster should not be accepted at all. This would fix components like the jet finder, but still allows misuse by the users. The momentum vector is in this case put in the middle of the acceptance and we might want to put it outside the ALICE acceptance.

File: PWG/EMCAL/EMCALbase/AliMCParticleContainer.cxx (hunk 1 above)
Comment (mfasDa) [reply PWG/EMCAL/EMCALbase/AliMCParticleContainer.cxx]: Anyhow planned once the review is over.

**Comment by lcunquei** [comment]
looks fine to me

**Comment by mfasDa** [comment]
+test

**Comment by mfasDa** [comment]
+1

**Comment by raymondEhlers** [comment]
Hi @mfasDa,

I seem to be having a problem with your change to the particle container title. See the reproduction below (the particle container doesn't work, while the cluster container works as expected)

~~~bash
root [0] AliParticleContainer test("tracks")
(AliParticleContainer &) Name: tracks Title: tracks_pT0150
root [1] test.SetParticlePtCut(3)
root [2] test.GetTitle()
(const char *) "tracks_pT0150" // Should be "tracks_pT3000"
root [3] AliClusterContainer test2("caloClusters")
(AliClusterContainer &) Name: caloClusters Title: caloClusters_E0000
root [8] test2.SetMinE(3)
root [9] test2.GetTitle()
(const char *) "caloClusters_E3000"
~~~

The title for the particle container should be "tracks_pT3000". We need to resolve this quickly, as it breaks many of my wagons on the LEGO train.

Best,
Raymond

**Comment by raymondEhlers** [comment]
I think I've fixed it. See #2973 .

**Pull request merged**